### ISSUE TITLE
Update Product with newer MSI silent

### ISSUE THREAD

**Issue by Torchok19081986** [oleg-shilo/wixsharp#456]
Hallo Oleg, i have one problem and hope you can help me with. I tried to execute msi silently with paramters /q, but msi doesnt update my product to newer version. if i run it with double click, here works fine and its all correct and newer version are being installed. My problem is, that i cant execute msi always directly, because there are too many servers/pcs for my software and i really need to update it for all of them. after msi being downloading from our space, ftp-server, i tried to start extern process with processtartingfo, but nothing happend. 
User and password are domain administrator and his password, also same domainname. I always thought, that i can do silently same as normal double click with difference, that i can update my product to newer version. Guid and productcode are always same. Is there some posibility to run msi silently ?
Update 13.08.18 09: 53 : 
I tried your solution in major upgrade from setup1.msi silent to setup2.msi. Here works fine. Do i unterstand corectly, that update msi shouldnt have any gui from execute update ?

**Comment by Torchok19081986** [comment]
I founded my mistake. It works not silent, if there onlydetect not definied and set to true und removaproduct not set to step. With this 2 properties works great. Only one Question: how do i search current installed version of my product and execute some custom actions ? registry keys are created in previous installation, but i dont know how to set it before installation begins. Is before_insall event are usefull?

**Comment by oleg-shilo** [comment]
Have a look at 'registry search' sample. Most likely it is what you are looking for.

**Comment by Torchok19081986** [comment]
Many thanks for quick answer Oleg. I looking in my registry for some info. In Sample of RegistrySearch is search for Sqlserverbrowser with exe, but i dont have a .exe file path in my registry. Only path to Installdir of my product, which is shared folder. I tried it with with RegistrySearch, but i got always empty path to my installdir. Code looks like :   
new Property("SEARCH_INSTALLDIR",
                new RegistrySearch(RegistryHive.LocalMachine, @"\SOFTWARE\Wow6432Node\Companyname\Myproduct\", "Installdir", RegistrySearchType.directory);
I got in Installdir_dialog empty string. Is something wrong in my code or do i need execute it as CustomAction ?
